fn/fn.go

```diff
@@ -23,7 +23,7 @@ func Iden[A any](a A) A {
 // Const is a function that accepts an argument and returns a function that
 // always returns that value irrespective of the returned function's argument.
 // This is also quite useful in conjunction with higher order functions.
-func Const[A, B any](a A) func(B) A {
+func Const[B, A any](a A) func(B) A {
```

Comment:
Unnecessary change?

Reply:
Not *necessary* but is a large QOL boost. The A parameter can be inferred by the supplied argument, but the B parameter cannot. Go does not require all type parameters to be specified if it can infer them from the arguments. However go *does NOT* possess the ability to do return type inference based off of the calling context. So by reversing the order of the type parameters here we now *only* need to specify B in most places since A will be inferred.

Reply:
TIL!